compare.py

```diff
@@ -0,0 +1,204 @@
+import os
+import numpy as np
+from scipy import signal
+from scipy.io import wavfile
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+from matplotlib.colors import LinearSegmentedColormap
+from matplotlib import cm
+
+
+class Compare:
+    """
+    Compare the output of the model to the source material.
+    ---
+    Generates data viz for freq response, power spectral density,
+    mel spectrogram diff, and waveform comparison.
+
+    Does not analyze metadata of the model file, which would be
+    interesting as an additional output.
+    """
+    def __init__(self, out_dir,  save=True, display=False):
+        self.save = save
+        self.display = display
+        if not save and not display:
+            raise ValueError("Plot is set to neither save nor display")
+        print((f"Generating visualization..."))
+        self.validate(f'models/{out_dir}')
+        self.analyze()
+
+    def validate(self, out_dir):
+        """
+        Confirm that the path and its expected children exist.
+        Valid results are stored on the class.
+        """
+        if not os.path.exists(out_dir):
+            raise ValueError(f"Directory {out_dir} does not exist")
+        else:
+            self.out_dir = out_dir
+        self.dry_wav = os.path.join(out_dir, 'x_test.wav')
+        self.train_wav = os.path.join(out_dir, 'y_test.wav')
+        self.pred_wav = os.path.join(out_dir, 'y_pred.wav')
+
+        if not os.path.exists(self.dry_wav):
+            raise ValueError(f"Input wav file {self.dry_wav} does not exist")
+        if not os.path.exists(self.train_wav):
+            raise ValueError(f"Output wav file {self.train_wav} does not exist")
+        if not os.path.exists(self.pred_wav):
+            raise ValueError(f"Prediction wav file {self.pred_wav} does not exist")
+        
+        # Load data, use a single known-good sample rate
+        sr_d, dry = wavfile.read(self.dry_wav)
+        sr_t, train = wavfile.read(self.train_wav)
+        sr_p, pred = wavfile.read(self.pred_wav)
+        if sr_d != sr_t or sr_t != sr_p:
+            raise ValueError("Sample rates of the wav files do not match. Are they from the same session?")
+        else:
+            self.sr = sr_p
+        
+        # File lengths can sometimes differ by a sample or two. This will even
+        # out a negligible difference but throw for anything noteworthy
+        shortest_data = min(len(dry), len(train), len(pred))
```

Comment:
Bumped into this issue with the test data. Wrote around it slightly doesn't seem problematic.